trinity/sync/common/headers.py

```diff
@@ -989,3 +1021,16 @@ def _is_syncing_skeleton(self) -> bool:
                 "%s announced Head TD %d, which is higher than ours (%d), starting sync",
                 peer, peer.head_info.head_td, head_td)
             pass
+
+    def _get_sync_status(self) -> Tuple[bool, Optional[SyncProgress]]:
+        if not self._is_syncing_skeleton or not self._meat.sync_progress:
+            return False, None
+        return True, self._meat.sync_progress
+
+    async def _handle_sync_status_requests(self) -> None:
+        event_bus = self._peer_pool.get_event_bus()
+        async for req in event_bus.stream(SyncingRequest):
```

Comment:
You want this loop to cancel out when `_build_skeleton()` exits, right? I think this will do what you're looking for:

```suggestion
        async for req in self.wait_iter(event_bus.stream(SyncingRequest)):
```

Reply:
Thank you very much @carver. I struggled a lot with this. `asyncio` is a very cryptic library 😆